Source/cairo/CairoFontInfo.m

```diff
@@ -71,16 +82,48 @@ - (BOOL) setupAttributes
       return NO;
     }
 
-  // We must not leave the hinting settings as their defaults,
-  // because if we did, that would mean using the surface defaults
-  // which might or might not use hinting (xlib does by default.)
-  //
-  // Since we make measurements outside of the context of a surface
-  // (-advancementForGlyph:), we need to ensure that the same
-  // hinting settings are used there as when we draw. For now,
-  // just force hinting to be off.
-  cairo_font_options_set_hint_metrics(options, CAIRO_HINT_METRICS_ON);
-  cairo_font_options_set_hint_style(options, CAIRO_HINT_STYLE_NONE);
+  if ((hinting = [ud integerForKey: @"GSFontHinting"]))
+    { /*
+       * This part is only here to debug disabling the use of hinting for
+       * metrics, because doing so causes menus to get cut off on the right.
+       */
+      switch (hinting >> 4)
+	{
+	  case 1:
+	    metrics = CAIRO_HINT_METRICS_OFF;
+	    break;
+	  case 2:
+	    metrics = CAIRO_HINT_METRICS_ON;
+	    break;
+	}
+
+      switch (hinting & 0x0f)
+	{
+	  case 1:
+	    style = CAIRO_HINT_STYLE_NONE;
+	    break;
+	  case 2:
+	    style = CAIRO_HINT_STYLE_SLIGHT;
+	    break;
+	  case 3:
+	    style = CAIRO_HINT_STYLE_MEDIUM;
+	    break;
+	  case 4:
+	    style = CAIRO_HINT_STYLE_FULL;
+	    break;
+	}
+    }
+  cairo_font_options_set_hint_metrics(options, metrics);
+  cairo_font_options_set_hint_style(options, style);
+
+  if ((subpixel = [ud integerForKey: @"back-art-subpixel-text"]))
```

Comment:
History. The default was added by Alex Malmberg in 2002 or so in the libart backend to do subpixel rendering of text. When he added the feature to back-art we publicized it, and as a result it's been added to user-facing software. So, despite extending support for the feature to Cairo, I felt it was inappropriate to arbitrarily rename the option or create a second one.

Reply:
No, instead we should add a new option, with a meaningful name. What we also should do is to use the old name as a fallback in the art backend, so that there both names get supported. Of course the new name also needs to be documented.